pilot/pkg/model/service.go

```diff
@@ -1768,3 +1788,13 @@ func (ep *IstioEndpoint) Equals(other *IstioEndpoint) bool {
 
 	return true
 }
+
+func EqualUsingPremarshaled[T proto.Message](a T, am *anypb.Any, b T, bm *anypb.Any) bool {
```

Comment:
Should this be in `pkg/proto`?

Reply:
Its kind of niche.. maybe Ill just unexport it?